mage_ai/data_preparation/models/block.py

```diff
@@ -540,11 +540,20 @@ def get_outputs(self, sample_count=None):
                 sample_count=sample_count,
             )
             if type(data) is pd.DataFrame:
+                analysis = variable_manager.get_variable(
+                    self.pipeline.uuid,
+                    self.uuid,
+                    v,
+                    variable_type=VariableType.DATAFRAME_ANALYSIS,
+                )
+                stats = analysis.get('statistics', {})
+                column_types = analysis.get('metadata', {}).get('column_types', {})
                 data = dict(
                     sample_data=dict(
                         columns=data.columns.tolist(),
                         rows=data.to_numpy().tolist(),
                     ),
+                    shape=[stats.get('count'), len(column_types)],
```

Comment:
we set the maximum row count to 100k when analyzing the dataframe. we might want to store the the accurate row count in dataframe analysis first.

Reply:
@wangxiaoyou1993 i updated it so that the original df is passed down into the statistics calculator. right now it's only used for calculating the row count, but i thought it could be useful for other calculations potentially